designer/server/src/createServer.ts

```diff
@@ -84,6 +84,16 @@ export async function createServer() {
   server.method('session.set', (id, value) => cache.set(id, value))
   server.method('session.drop', (id) => cache.drop(id))
 
+  server.method('state.get', (userId, key) => {
+    if (!userId) return Promise.reject(new Error('userId is not defined'))
```

Comment:
Can I check when the `userId` might be empty?

If you've found a way this is possible we should update `hasUser()` instead

Otherwise can probably remove the checks

Reply:
I am not sure in real scenario but for testing I have removed auth and userId doesn't exist in that case

Reply:
I'm trying to avoid writing code we can't test and is likely never run

E.g. With the [suggestion above](https://github.com/DEFRA/forms-designer/pull/509#discussion_r1783976070) is it definitely possible to have no user ID?

```js
if (!hasUser(credentials)) {
  return Boom.unauthorized()
}
```

Reply:
after adding above Boom return we don't need error in createserver.js, I will remove